WebGL: Fix texStorage2D target validation

lexaknyazev opened the pull request:
#### 78f7319cdcb5670eb247d033fe00d5c1edba29de
<pre>
WebGL: Fix texStorage2D target validation
<a href="https://bugs.webkit.org/show_bug.cgi?id=237427">https://bugs.webkit.org/show_bug.cgi?id=237427</a>

Reviewed by Kenneth Russell.

Added a new validation helper because the list of allowed target
enums for texStorage2D does not match other texture entry points.

The validation should be removed, as same is done in ANGLE,
but for consistency it should be removed from all functions.

Until removal, use correct WebKit-side validation.

See deqp/functional/gles3/negativetextureapi.html

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::validateTextureStorage2DBinding):
(WebCore::WebGL2RenderingContext::texStorage2D):
* Source/WebCore/html/canvas/WebGL2RenderingContext.h:

Canonical link: <a href="https://commits.webkit.org/252896@main">https://commits.webkit.org/252896@main</a>
</pre>

kenrussell reviewed (approved):
Looks good to me. r+

lexaknyazev committed:
WebGL: Fix texStorage2D target validation
https://bugs.webkit.org/show_bug.cgi?id=237427

Reviewed by Kenneth Russell.

Added a new validation helper because the list of allowed target
enums for texStorage2D does not match other texture entry points.

The validation should be removed, as same is done in ANGLE,
but for consistency it should be removed from all functions.

Until removal, use correct WebKit-side validation.

See deqp/functional/gles3/negativetextureapi.html

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::validateTextureStorage2DBinding):
(WebCore::WebGL2RenderingContext::texStorage2D):
* Source/WebCore/html/canvas/WebGL2RenderingContext.h:

Canonical link: https://commits.webkit.org/252896@main
